pkg/ctl/functions/util.go

```diff
@@ -275,14 +275,15 @@ func validateFunctionConfigs(functionConfig *util.FunctionConfig) error {
 		utils.InferMissingNamespace(functionConfig)
 	}
 
-	if functionConfig.Jar != nil && functionConfig.Py != nil && functionConfig.Go != nil {
```

Comment:
This error check was replaced because:

1. It will only fire if all three flags are provided, not just a mixture of two.
2. The error says "both", so, just like the above point mentiones, it's intended only to address an older case where there were only two kinds of functions.

The more generic method using `NumProvidedStrings` allows a more sophisticated check that can assert that the user provided one (and only one) flag referencing the language of the pulsar function, as per expected usage. It will now work no matter how many languages are added .